code/controllers/subsystem/processing/processing.dm

```diff
@@ -22,9 +22,7 @@ SUBSYSTEM_DEF(processing)
 	while(current_run.len)
 		var/datum/thing = current_run[current_run.len]
 		current_run.len--
-		if(thing)
-			thing.process(wait)
-		else
+		if(!QDELETED(thing) && thing.process(wait) != PROCESS_KILL)
```

Comment:
technically shouldn't anything that returns `TRUE` on either `QDELETED` or `PROCESS_KILL` be removed from the processing list?